engine/src/main/java/org/terasology/particles/updating/ParticleUpdaterImpl.java

```diff
@@ -116,9 +118,10 @@ public void removeEmitter(final EntityRef emitter) {
     public void update(final float delta) {
         movingAvgDelta = TeraMath.lerp(movingAvgDelta, delta, 0.05f);
 
-        for (ParticleEmitterComponent registeredParticleSystem : ImmutableList.copyOf(registeredParticleSystems)) {
-            updateParticleSystem(registeredParticleSystem, delta);
-        }
+        Collection<ParticleEmitterComponent> particleEmitters = ImmutableList.copyOf(registeredParticleSystems);
+        particleEmitters.forEach(x -> updateParticleEmitters(x, delta));
+        particleEmitters.forEach(x -> updateParticleData(x, delta));
```

Comment:
Is it important that particle emitters and particale data are updated in separate loops, i.e., that the particle data is updated after _all_ emitters have been updated? If so, please provide a brief comment stating the reason here.

Otherwise, this could be done in a single `forEach`:
```suggestion
        particleEmitters.forEach(x -> {
            updateParticleEmitters(x, delta);
            updateParticleData(x, delta);
        });
```

Reply:
comment suggestion:
```
// It's important to update all emitters before the particle data inside the pools gets updated.
// This ensures that all freshly revived particles are also being updated.
```
To explain it further:
In the code you suggested a particle pool would be updated and in following loop iterations, other emitters might revive additional particles in that pool. These weren't part of the update.

Please provide feedback if the comment makes that clear enough. I'll add it then.

Reply:
It does, thanks for explaining! 👍